### PR TITLE
Create "image_compare_threshold" Argument

### LINKED ISSUE #366: Add Command Line Argument that supports Image Compare Thresholds
# Requested Update

This comes from a conversation had by @mindyls, @jsolbrig, and myself.

### Description
Currently, the output_checkers / compare_outputs.py image comparison methods use a static threshold to compare images. This threshold, which is the percent difference allowed between two pixels, is used to allow pixels that are "close enough" to pass, however we don't always want that to occur. Specifically, we should allow three different threshold values ["lenient", "medium", "strict"], so we can control the preciseness of our image comparisons.

### Background and Motivation
We don't always want a lenient threshold, and having this argument allowed for image comparisons will solve this problem. 

### Alternative Solutions
We can manually set this value. Not a good solution however.

### Code to demonstrate issue
compare_outputs.py --> images_match
output_checkers/image.py --> outputs_match

### Checklist for Completion
* [ ] Create an argument for "image_compare_threshold".
* [ ] Link this argument throughout relevant code.

### PR DESCRIPTION
**Pull request by evrose54**
# Reviewer Checklist

* [x] Required ***existing tests*** pass (ie full_test.sh, others as appropriate)
  * [x] None required (explain if checked, delete line if not checked)
* [x] Required ***unit tests*** added and pass for new/modified functionality
  * [x] None required (explain if checked, delete line if not checked)
* [x] Required ***integration tests*** added and pass for new/modified functionality
  * [x] None required. (explain if checked, delete line if not checked)
* [x] Required ***documentation*** added for new/modified functionality
  * [x] None required (explain if checked, delete line if not checked)
* [x] Required ***release notes*** added for new/modified functionality
  * [x] None required. (explain if checked, delete line if not checked)

# Related Issues

fixes NRLMMD-GEOIPS/geoips#366
Based off of NRLMMD-GEOIPS/geoips#367

# Testing Instructions
Run geoips/tests/integration_tests/full_test.sh. An output of 0 means this ran correctly.

# Summary

GeoIPS needs to add functionality for image comparisons, as well as other types of comparisons. Specifically for image comparisons, we need to allow for different thresholds between pixels. This threshold argument (the % difference between pixels), will benefit users comparing images with different needs of accuracy. We will implement this via "output_checker_kwargs"

    - modified: data_fusion/data_fusion/plugins/modules/procflows/data_fusion.py

### CODE REVIEW

File: data_fusion/plugins/modules/procflows/data_fusion.py [hunk 1]
```diff
@@ -529,6 +529,11 @@ def call(fnames, command_line_args=None):
         command_line_args, {"METADATA": fuse_data["final"]["metadata_xobj"]}
     )
 
+    if command_line_args.get("image_compare_threshold") is not None:
+        image_compare_threshold = command_line_args["image_compare_threshold"]
+    else:
+        image_compare_threshold = None
+
```
Comment (evrose54) [inline data_fusion/plugins/modules/procflows/data_fusion.py:533]: I noticed when working on this PR, arguments "logging_level" and "no_presectoring" are not checked for. We might want to create an issue for these to be added in the future, and make the appropriate changes to the code.

Comment (mindyls) [reply data_fusion/plugins/modules/procflows/data_fusion.py:533]: https://github.com/NRLMMD-GEOIPS/data_fusion/issues/26
Created an issue.  Resolving this!

**Review by mindyls — CHANGES_REQUESTED** [review]
Like in the geoips repo, I think we should update this to take "output_checker_kwargs" command line, then we can just pass in arbitrary kwargs.

File: data_fusion/plugins/modules/procflows/data_fusion.py [hunk 2]
```diff
@@ -529,6 +529,11 @@ def call(fnames, command_line_args=None):
         command_line_args, {"METADATA": fuse_data["final"]["metadata_xobj"]}
     )
 
+    if command_line_args.get("image_compare_threshold") is not None:
+        image_compare_threshold = command_line_args["image_compare_threshold"]
+    else:
+        image_compare_threshold = None
```
Comment (mindyls) [inline data_fusion/plugins/modules/procflows/data_fusion.py]: Replace with output_checker_kwargs.  Will be like  output_checker_kwargs["image"]["threshold"] = "medium"

File: data_fusion/plugins/modules/procflows/data_fusion.py [hunk 3]
```diff
@@ -529,6 +529,16 @@ def call(fnames, command_line_args=None):
...
 
+    if command_line_args.get("output_checker_kwargs") is not None:
+        output_checker_kwargs = command_line_args["output_checker_kwargs"]
+    else:
+        output_checker_kwargs = {
+            "image": {"image_threshold": "medium"},
+            "geotiff": {},
+            "netcdf": {},
+            "text": {},
+        }
```
Comment (mindyls) [inline data_fusion/plugins/modules/procflows/data_fusion.py]: output_checker_kwargs = command_line_args.get("output_checker_kwargs", {})
Remove explicit plugin names.

Comment (jsolbrig) [reply data_fusion/plugins/modules/procflows/data_fusion.py]: It looks like this still needs to be resolved, correct? @evrose54 @mindyls

Comment (evrose54) [reply data_fusion/plugins/modules/procflows/data_fusion.py]: This has been updated now @jsolbrig @mindyls

File: data_fusion/plugins/modules/procflows/data_fusion.py (hunk 2 above)
Comment (jsolbrig) [reply data_fusion/plugins/modules/procflows/data_fusion.py]: It looks to me like this has been fixed. Can one of you confirm @mindyls @evrose54 ?

**Review by jsolbrig — APPROVED** [review]

File: data_fusion/plugins/modules/procflows/data_fusion.py (hunk 2 above)
Comment (mindyls) [reply data_fusion/plugins/modules/procflows/data_fusion.py]: Is this tested/used at all @evrose54  ?  I can't remember the status

**Review by mindyls — APPROVED** [review]
Assuming this is used in at least one test, I am good with this.

File: data_fusion/plugins/modules/procflows/data_fusion.py (hunk 2 above)
Comment (evrose54) [reply data_fusion/plugins/modules/procflows/data_fusion.py]: It's been fixed. It's not used at all to my knowledge however.

**Pull request merged**